larrecodnn/ImagePatternAlgs/Tensorflow/Modules/PointIdTrainingData_module.cc

```diff
@@ -206,6 +332,122 @@ namespace nnet {
           writeAndDelete(pdgHist);
 
         }
+        else if (fDumpToNumpy){
+          for (size_t w = w0; w < w1; ++w) {
+            int w_start = w - fPatch_size_w/2;
+            int w_stop = w_start + fPatch_size_w;
+            if (w_start < int(w0) || w_start > int(w1)) continue;
+            if (w_stop < int(w0) || w_stop > int(w1)) continue;
+            auto const& pdg = fTrainingDataAlg.wirePdg(w);
+            auto const& deposit = fTrainingDataAlg.wireEdep(w);
+            auto const& raw = fTrainingDataAlg.wireData(w);
+            for (size_t d = d0; d < d1; ++d) {
+              int d_start = d - fPatch_size_d/2;
+              int d_stop = d_start + fPatch_size_d;
+              if (d_start < int(d0) || d_start > int(d1)) continue;
+              if (d_stop < int(d0) || d_stop > int(d1)) continue;
+              int y0 = 0, y1 = 0, y2 = 0, y3 = 0;
+              if (deposit[d]<2e-5 || raw[d]<0.05){//empty pixel
+                y3 = 1;
+                ++nNone;
+                if (flat.fire()>fNone) continue;
+                ++nNone_sel;
+              }
+              else if ((pdg[d] & 0x0FFF) == 11){//shower
+                y1 = 1;
+                ++nEm;
+                if ((pdg[d] & 0xF000) == 0x2000){//Michel
+                  y2 = 1;
+                  ++nMichel;
+                  if (flat.fire()>fMichel) continue;
+                  ++nMichel_sel;
+                }
+                else{
+                  if (flat.fire()>fEm) continue;
+                  ++nEm_sel;
+                }
+              }
+              else{//track
+                y0 = 1;
+                ++nTrk;
+                //Check if the track appears to be stopping
+                int nPxlw0 = 0, nPxlw1 = 0, nPxld0 = 0, nPxld1 = 0;
+                //Try to fit a track
+                std::vector<double> wfit, dfit, chgfit;
+                double total_trkchg = 0;
+                for (int ww = w_start; ww < w_stop; ++ww){
+                  auto const& pdg1 = fTrainingDataAlg.wirePdg(ww);
+                  auto const& deposit1 = fTrainingDataAlg.wireEdep(ww);
+                  auto const& raw1 = fTrainingDataAlg.wireData(ww);
+                  for (int dd = d_start; dd < d_stop; ++dd){
+                    if (deposit1[dd]<2e-5 || raw1[dd]<0.05) continue; //empty pixel
+                    if ((pdg1[dd] & 0x0FFF) != 11){//track pixel
+                      wfit.push_back(ww);
+                      dfit.push_back(dd);
+                      chgfit.push_back(raw1[dd]);
+                      total_trkchg += raw1[dd];
+                      if (ww-w_start < 3) ++nPxlw0;
+                      if (w_stop-ww-1 < 3) ++nPxlw1;
+                      if (dd-d_start < 3) ++nPxld0;
+                      if (d_stop-dd-1 < 3) ++nPxld1;
+                    }
+                  }
+                }
+                //Fit track pixels
+                double fit_trkchg = 0;
+                double parm[2];
```

Comment:
Suggest initializing to `0.` right away to avoid having to do it in the `WeightedFit` function:

```suggestion
                double parm[2] = {};
```